src/AppBundle/QueryType/MenuQueryType.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace AppBundle\QueryType;
+
+use eZ\Publish\Core\QueryType\QueryType;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+class MenuQueryType implements QueryType
+{
+    /** @var string[] */
+    private $languages;
+
+    /**
+     * @param string[] $value
+     */
+    public function setLanguages($value)
+    {
+        $this->languages = $value;
+    }
+
+    public function getQuery(array $parameters = [])
+    {
+        $criteria = new Query\Criterion\LogicalAnd([
+            new Query\Criterion\Visibility(Query\Criterion\Visibility::VISIBLE),
+            new Query\Criterion\Location\Depth(
+                Query\Criterion\Operator::BETWEEN,
+                [
```

Comment:
No need to wrap this array's content on multiple lines, it will be as readable on one line. Importing `eZ\Publish\API\Repository\Values\Content\Query\Criterion` in the current namespace would also allow you to shorten this down:

```
$criteria = new Criterion\LogicalAnd([
  new Criterion\Visibility(Criterion\Visibility::VISIBLE),
  new Criterion\Location\Depth(Criterion\Operator::BETWEEN, [1, 2]),
  new Criterion\Subtree('/1/2/'),
  new Criterion\LanguageCode($this->languages),
```

Also, could the depth be handled as `new Criterion\Location\Depth(Criterion\Operator::LTE, 2)` ?